File: tests/drawdown_test.py
import numpy as np

from taflow import Drawdown


def test_running_drawdown_and_reset() -> None:
    values = np.array([10.0, 8.0, 12.0, 9.0])
    state = Drawdown(values)
    np.testing.assert_allclose(state.compute(), [0.0, -0.2, 0.0, -0.25])
    assert state.reset().value is None

File: crates/taflow-core/src/stream/drawdown_test.rs
use super::drawdown::Drawdown;

#[test]
fn running_maximum_and_reset_are_consistent() {
    let mut state = Drawdown::new();
    assert_eq!(state.append(10.0), 0.0);
    assert!((state.append(8.0) + 0.2).abs() < 1e-12);
    state.reset();
    assert_eq!(state.value(), None);
    assert_eq!(state.append(10.0), 0.0);
}
